src/components/FilmList/FilmList.tsx

```diff
@@ -1,28 +1,58 @@
+import { useEffect, useState } from "react";
+import { filterByGenre, sorting } from "../../helpers";
 import { Film } from "../../interfaces";
 import { FilmCard } from "../FilmCard";
 
 import "./style.scss";
 
 interface FilmListProps {
+  selectedGenre: string;
+  sortBy: string;
+  setOpenedFilmId(filmID: string): void;
   movies: Film[];
   openModal(type: string, filmID: string): void;
 }
 
-const FilmList: React.FC<FilmListProps> = ({ movies, openModal }) => {
-  const filmsCards: any = movies.map((el) => {
-    return <FilmCard film={el} key={el.id} openModal={openModal} />;
-  });
+const FilmList: React.FC<FilmListProps> = ({
+  movies,
+  openModal,
+  selectedGenre,
+  sortBy,
+  setOpenedFilmId,
+}) => {
+  const [moviesConfigs, setMoviesConfigs] = useState(movies);
+
+  useEffect(() => {
+    setMoviesConfigs((prevState) => {
+      return [...sorting(prevState, sortBy)];
+    });
+  }, [sortBy]);
+
+  useEffect(() => {
+    setMoviesConfigs(filterByGenre(movies, selectedGenre));
+  }, [movies, selectedGenre]);
```

Comment:
wrong logic. in case genre change, after sort action, you only filter by genre base movies, which wouldn't be sorted by current sortBy.